docs/advanced/platform-bom.md

```diff
@@ -0,0 +1,141 @@
+Using the Platform Bill of Materials (BOM)
+
```

Comment:
This is assuming all users understand why we need to use a BOM. It would be good to add a few lines explaining with an example the problem this prevents.

Reply:
done.